### PR TITLE
Add cmark package

### PR DESCRIPTION
**Pull request by charlesseizilles**

### CODE REVIEW

File: packages/c/cmark/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package("cmark")
+    set_kind("library")
```
Comment (star-hengxing) [inline packages/c/cmark/xmake.lua:2]: package default kind is library, we need't add it

File: packages/c/cmark/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        local configs = {
+            "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"),
+            "-DBUILD_SHARED_LIBS=" .. (package:config("shared") and "ON" or "OFF"),
+            "-DBUILD_TESTING=OFF",
+        }
+        import("package.tools.cmake").install(package, configs)
+    end)
+
+    on_test(function (package)
+        io.read()
```
Comment (star-hengxing) [inline packages/c/cmark/xmake.lua:21]: Please remove unused code

File: packages/c/cmark/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package("cmark")
+    set_homepage("https://github.com/commonmark/cmark")
+    set_description("CommonMark parsing and rendering library and program in C")
+
+    add_urls("https://github.com/commonmark/cmark.git")
+    add_versions("0.31.0", "8fbf029685482827828b5858444157052f1b0a5f")
```
Comment (star-hengxing) [inline packages/c/cmark/xmake.lua:6]: Why not add [targz](https://github.com/commonmark/cmark/archive/refs/tags/0.31.0.tar.gz) archive?

Comment (charlesseizilles) [reply packages/c/cmark/xmake.lua:6]: no particular reason, I can add it if you want

Comment (star-hengxing) [reply packages/c/cmark/xmake.lua:6]: We can use `xmake l scripts/new.lua github:commonmark/cmark` to automatically generate package, which will save some time

Comment (charlesseizilles) [reply packages/c/cmark/xmake.lua:6]: oh I didn't know about that!

**Pull request merged**